AsyncDisplayKit/TextKit/ASTextKitRenderer.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <vector>
+
+#import <UIKit/UIKit.h>
+
+#import "ASTextKitAttributes.h"
+
+@class ASTextKitContext;
+@class ASTextKitShadower;
+@protocol ASTextKitTruncating;
+
+/**
+ ASTextKitRenderer is a modular object that is responsible for laying out and drawing text.
+
+ A renderer will hold onto the TextKit layouts for the given attributes after initialization.  This may constitute a
+ large amount of memory for large enough applications, so care must be taken when keeping many of these around in-memory
+ at once.
+
+ This object is designed to be modular and simple.  All complex maintenance of state should occur in sub-objects or be
+ derived via pure functions or categories.  No touch-related handling belongs in this class.
+
+ ALL sizing and layout information from this class is in the external coordinate space of the TextKit components.  This
+ is an important distinction because all internal sizing and layout operations are carried out within the shadowed
+ coordinate space.  Padding will be added for you in order to ensure clipping does not occur, and additional information
+ on this transform is available via the shadower should you need it.
+ */
+@interface ASTextKitRenderer : NSObject
+
+/**
+ Designated Initializer
+dvlkferufedgjnhjjfhldjedlunvtdtv
```

Comment:
lol, did I write this on the original diff? Must have leaned on my keyboard or somehting :P